mods/persistence/modules/mob/new_player.dm

```diff
@@ -155,3 +151,56 @@
 /mob/new_player/close_spawn_windows()
 	close_browser(src, "window=latechoices") //closes late choices window
 	panel.close()
+/mob/new_player/proc/AttemptLateSpawnOutreach(var/datum/job/job, var/spawning_at)
+	if(src != usr)
+		return 0
+
+	if(GAME_STATE != RUNLEVEL_GAME)
+		to_chat(usr, "<span class='warning'>The round is either not ready, or has already finished...</span>")
+		return 0
+
+	if(!config.enter_allowed)
+		to_chat(usr, "<span class='notice'>There is an administrative lock on entering the game!</span>")
+		return 0
+
+	var/decl/spawnpoint/spawnpoint = job.get_spawnpoint(client)
+	if(!spawnpoint)
+		to_chat(src, alert("That spawnpoint is unavailable. Please try another."))
+		return 0
+
+	var/turf/spawn_turf = pick(spawnpoint.turfs)
+	if(job.latejoin_at_spawnpoints)
+		var/obj/S = job.get_roundstart_spawnpoint()
+		spawn_turf = get_turf(S)
+
+	if(!SSjobs.check_unsafe_spawn(src, spawn_turf))
+		return
+
+	SSjobs.assign_role(src, job.title, 1)
+
+	var/mob/living/character = create_character(spawn_turf)	//creates the human and transfers vars and mind
+	if(!character)
+		return 0
+	if(character.mind)
+		SSpersistence.AddToLimbo(character.mind, character.mind.unique_id, LIMBO_MIND, character.mind.key, TRUE)
```

Comment:
You don't need to copy paste and reimplement the whole proc just for this line. If you look at the spawnpoint code (/decl/spawnpoint/chargen/after_join()), you can have the character save after there instead. It'll make a lot less problem in the future for us.